py-pkg/pkgnet/dependency_reporter.py

```diff
@@ -0,0 +1,63 @@
+from typing import Optional, Callable
+
+try:
+    import importlib.resources as importlib_resources
+except ImportError:
+    # Python 3.6 uses importlib_resources backport
+    import importlib_resources
+
+
+import pandas as pd
+import pkg_resources
+
+from pkgnet.abstract_graph_reporter import AbstractGraphReporter
+from pkgnet.abstract_package_reporter import registrar
+from pkgnet.graphs import DirectedGraph
+from pkgnet.search_utils import recursive_node_search
+
+
+@registrar.register_reporter
+class DependencyReporter(AbstractGraphReporter):
+
+    _graph_class = DirectedGraph
+
+    report_template = "tab_dependency_report.jinja"
+    report_slug = "dependency-report"
+    report_name = "Dependencies"
+    layout = "kamada_kawai_layout"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._ignore_packages = []
+
+    ### PUBLIC METHODS ###
+
+    @classmethod
+    def report_template(cls) -> (str, str, Optional[Callable]):
+        source = importlib_resources.read_text("pkgnet.templates", "tab_dependency_report.jinja")
+        path = next(importlib_resources.path("pkgnet.templates", "tab_dependency_report.jinja").gen)
+        mtime = path.stat().st_mtime  # last modified time
+        return source, str(path), lambda: path.stat().st_mtime == mtime
```

Comment:
in my opinion, returning more than 2 things makes it hard to reason about a method and makes mistakes easier. What do you think about returning a custom object or a dictionary instead?

Reply:
This is what `jinja2` loaders expect to be returned, and not something I designed. https://jinja.palletsprojects.com/en/2.11.x/api/#jinja2.BaseLoader